### ISSUE TITLE
digraph insertion replaces next character temporarily

### ISSUE THREAD

**Issue by mqudsi** [neovim/neovim#10315]
- `nvim --version`: v0.4.0-474-gdfb7f6b34
- Vim (version: 8.1) behaves differently? no
- Operating system/version: Ubuntu
- Terminal name/version: conhost 18362
- `$TERM`: xterm-256color

### Steps to reproduce using `nvim -u NORC`

```
nvim -u NORC
ix←^k
```

### Actual behaviour
The previously entered `x` is overwritten with the digraph-in-progress `?` 

### Expected behaviour
The `?` should be inserted where the new digraph is expected to be inserted, rather than overwriting the next character.  In my case, the digraph I was inserting was based off of the next character, which I could no longer see.

**Comment by justinmk** [comment]
This would cause the text to shift, and then if the digraph is canceled the text shifts back.

**Comment by bfredl** [comment]
Possible alternative: don't display any `?`, and indicate the waiting state by some other means, like changed cursor shape.

**Comment by mqudsi** [comment]
> This would cause the text to shift. [snip]

Yes, but in the general case, the shift will be happening anyway.

> And the text will shift twice if 
> the digraph is canceled
> the final digraph occupies 2 cells

Yes, but not in an unexpected manner or in a jerky way that makes it an unpleasant ux. It's in response to user actions each time (so there's a "long" delay in comparison to flashes of content, i.e. human timescale not machine timescale). 

I would argue that shifting each time is the correct behavior, as at each point what is on screen is a more "correct" snapshot of the "state" of the document. We don't worry about shifting when we reach the end of a line because it's what the user *expects* to happen (and would be surprised if it didn't), imho we should treat this the same way.
